From a646aace8bfd6fc83e360c2fd0c406812922ba5f Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Wed, 13 Dec 2017 17:14:58 -0600
Subject: [PATCH 13/15] MLK-17205-3 video: imx: hdp: Adding configs for HDP
 firmware loading

Added default environment for hdp loading.
Added hdp loading as default for iMX8QM MEK board.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
(cherry picked from downstream commit 432e5b4347841095c3b5a8a0d106f35deadd006e)
(cherry picked from downstream commit cb78a32fa8eb8c37932be003ebe4fa1f8c46c1d5)
---
 configs/imx8qm_mek_defconfig | 1 +
 include/configs/imx8qm_mek.h | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/configs/imx8qm_mek_defconfig b/configs/imx8qm_mek_defconfig
index 1c67b98c5d..5aae2fd0d9 100644
--- a/configs/imx8qm_mek_defconfig
+++ b/configs/imx8qm_mek_defconfig
@@ -74,4 +74,5 @@ CONFIG_SPL_DM_REGULATOR_GPIO=y
 CONFIG_DM_SERIAL=y
 CONFIG_FSL_LPUART=y
 CONFIG_SPL_TINY_MEMSET=y
+CONFIG_VIDEO_IMX_HDP_LOAD=y
 # CONFIG_EFI_LOADER is not set
diff --git a/include/configs/imx8qm_mek.h b/include/configs/imx8qm_mek.h
index d06ed61c80..32fcf7eb8f 100644
--- a/include/configs/imx8qm_mek.h
+++ b/include/configs/imx8qm_mek.h
@@ -80,7 +80,11 @@
 		"source\0" \
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"hdp_addr=0x84000000\0" \
+	"hdp_file=hdmitxfw.bin\0" \
+	"loadhdp=fatload mmc ${mmcdev}:${mmcpart} ${hdp_addr} ${hdp_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
+		"if run loadhdp; then; hdp load ${hdp_addr}; fi;" \
 		"run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
 			"if run loadfdt; then " \
@@ -101,6 +105,7 @@
 		"else " \
 			"setenv get_cmd tftp; " \
 		"fi; " \
+		"if ${get_cmd} ${hdp_addr} ${hdp_file}; then; hdp load ${hdp_addr}; fi;" \
 		"${get_cmd} ${loadaddr} ${image}; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
 			"if ${get_cmd} ${fdt_addr} ${fdt_file}; then " \
-- 
2.13.6

